src/__tests__/events.js

```diff
@@ -0,0 +1,287 @@
+import React from 'react'
+import {mount, unmount} from './helpers/react'
+import {fireEvent} from '../'
+
+describe('Clipboard Events', () => {
+  ;['copy', 'paste'].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('input')
```

Comment:
This code is very similar throughout the test file. Could we extract it to a "testEvent" function?

```javascript
function testEvent({eventName, elementType}) {
  const node = document.createElement(elementType)
  const spy = jest.fn()
  node.addEventListener(eventName.toLowerCase(), spy)
  fireEvent[eventName](node)
  expect(spy).toHaveBeenCalledTimes(1)
}
```

Then most of the tests could be dramatically simplified to something like:

```javascript
describe('Clipboard Events', () => {
  ;['copy', 'paste'].forEach(
    eventName => {
      it(`fires ${eventName}`, () => {
        testEvent({eventName, elementType: 'input'})
      })
    },
  )
})
```

**Alternatively** it may even not be a bad idea to abstract it further:

```javascript
const eventTypes = [
  {type: 'Clipboard', events: ['copy', 'paste'], elementType: 'input'},
  {
    type: 'Composition',
    events: ['compositionEnd', 'compositionStart', 'compositionUpdate'],
    elementType: 'input',
  },
  // etc...
]

eventTypes.forEach(({type, events, elementType}) => {
  describe(`${type} Events`, () => {
    events.forEach(eventName => {
      it(`fires ${eventName}`, () => {
        const node = document.createElement(elementType)
        const spy = jest.fn()
        node.addEventListener(eventName.toLowerCase(), spy)
        fireEvent[eventName](node)
        expect(spy).toHaveBeenCalledTimes(1)
      })
    })
  })
})
```

I think doing this will make it easier to add cases and know which tests are for one-off situations :+1: It will also make this file shorter probably :)